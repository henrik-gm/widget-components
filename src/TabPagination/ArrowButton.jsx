import React, { PropTypes } from 'react';
import styles from './ArrowButton.scss';

const ArrowButton = ({ type, onClick }) => (
   <button className={styles[type]} onClick={onClick}>
      <svg className={styles.icon} width="12px" height="20px" viewBox="0 860 420 500" version="1.1" xmlns="http://www.w3.org/2000/svg">
         <path
            id="arrowLeft"
            transform="scale(1, -1) translate(0, -1500)"
            d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714
               5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286
               266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714
               13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
         />
      </svg>
   </button>
);

ArrowButton.propTypes = {
   /**
    * Arrow direction - left or right
    */
   type: PropTypes.oneOf(['left', 'right']).isRequired,

   /**
    * On click handler
    */
   onClick: PropTypes.func.isRequired
};

export default ArrowButton;
