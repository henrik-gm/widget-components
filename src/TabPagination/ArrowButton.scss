:local(.left),
:local(.right) {
   position: absolute;
   top: 0;
   bottom: 0;
   width: 35px;
   border: none;
   background: none;
   font-size: 15px;
   font-weight: bold;
   cursor: pointer;
   outline: none;
   padding: 0;

   :local(.icon) {
      fill: #000;
      opacity: .3;
      transition: opacity .3s;
   }

   &:hover :local(.icon) {
      opacity: 1;
   }
}

:local(.left) {
   background: linear-gradient(90deg, #fff 50%, hsla(0,0%,100%,0));
   left: 16px;
   text-align: left;
}

:local(.right) {
   background: linear-gradient(270deg, #fff 50%, hsla(0,0%,100%,0));
   right: 16px;
   text-align: right;

   :local(.icon) {
      transform: scale(-1, 1);
   }
}
