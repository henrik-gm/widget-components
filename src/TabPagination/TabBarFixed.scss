@import '~kambi-widget-core-library/src/scss/core.scss';

:local(.bar) {
   @include flexbox;

   list-style: none;
   margin: 0;
   padding: 0 0 3px 0;

   :local(.tab) {
      @include flex(1);

      cursor: pointer;
      position: relative;

      :local(.border) {
         position: absolute;
         bottom: -3px;
         height: 3px;
         left: 0;
         right: 0;
         visibility: hidden;
      }

      &:hover :local(.border),
      &.selected :local(.border) {
         visibility: visible;
      }
   }
}
