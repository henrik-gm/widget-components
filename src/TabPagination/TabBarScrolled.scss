:local(.container) {
   position: relative;
   padding: 0 16px;

   &.mobile-device {
      padding: 0;
   }
}

// field of view
:local(.eyeshot) {
   overflow-x: hidden;

   .mobile-device & {
      overflow-x: scroll;
   }
}

// tab bar
:local(.bar) {
   display: flex;
   margin: 0;
   transition: transform .3s;

   // let tab containers manage their width
   & > * {
      flex: 0 0 auto;
   }
}
