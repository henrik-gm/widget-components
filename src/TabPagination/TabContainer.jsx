import React, { PropTypes } from 'react';
import styles from './TabContainer.scss';

/*
 * Returns DOM element's width (in pixels)
 * @param {HTMLElement} el DOM element
 * @returns number|null
 */
const getWidth = function(el) {
   return el ? el.offsetWidth : null;
};

const TabContainer = ({ children, selected, onClick, onWidth }) => (
   <div
      className={[styles.tab, selected ? 'selected' : ''].join(' ')}
      onClick={onClick}
      ref={onWidth ? el => onWidth(getWidth(el)) : undefined}
   >
      {children}
      <i className={['KambiWidget-primary-background-color', styles.border].join(' ')} />
   </div>
);

TabContainer.propTypes = {
   /*
    * On click handler
    */
   onClick: PropTypes.func,

   /*
    * Is this tab currently selected?
    */
   selected: PropTypes.bool,

   /*
    * Tab contents
    */
   children: PropTypes.node,

   /*
    * Called when tab width is known
    */
   onWidth: PropTypes.func
};

export default TabContainer;
