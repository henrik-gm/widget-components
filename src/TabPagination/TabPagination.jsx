import React, { Children, Component, PropTypes } from 'react';
import ScrolledList from '../List/ScrolledList/ScrolledList';

/**
 * Tab Pagination component.
 * Component should receive list of tab contents as children.
 * It will render upper tab bar (for switching tabs).
 * Once a tab is clicked a tab content will be switched.
 * @memberOf widget-components
 */
class TabPagination extends Component {

   /*
    * Constructs.
    * @param {object} props Component properties
    */
   constructor(props) {
      super(props);

      this.state = {
         selected: this.props.selected
      };

      this.switchTab = this.switchTab.bind(this);
   }

   /*
    * Switches to given tab
    * @param {number} idx Tab index
    */
   switchTab(idx) {
      this.setState({ selected: idx });
   }

   /*
    * Creates TabPagination markup
    * @returns {XML}
    */
   render() {
      const children = Children.toArray(this.props.children);

      return (
         <div>
            {this.props.renderTabList({
               selected: this.state.selected,
               onItemClick: this.switchTab,
               children: this.props.children.map((a, i) => this.props.renderTab(i))
            })}
            <div>
               {children[this.state.selected]}
            </div>
         </div>
      );
   }
}

/**
 * Renders tab for given index. The tab will be placed inside upper tab bar.
 * @callback TabPagination_RenderTab
 * @param {number} idx Tab index
 * @returns ReactElement
 * @example
 *    idx => <div style={{width: 50, height: 50}}>Tab <strong>#{idx}</strong></div>
 */

/**
 * Renders upper tab list.
 * @callback TabPagination_RenderTabList
 * @param {TabPagination_RenderTabListArgs} args Contains properties which will control tab bar behaviour
 * @returns ReactElement
 *
 * @example <caption>Using custom tab bar component</caption>
 *    ({selected, onItemClick, children}) =>
 *       <CustomTabBar
 *          selected={selected}
 *          onItemClick={onItemClick}>
 *             {children}
 *       </CustomTabBar>
 *
 * @example <caption>Shorthand syntax can be used once function arguments and component properties names match.</caption>
 *    args => <CustomTabBar {...args}>{args.children}</CustomTabBar>
 *
 * @example <caption>Usage of build-in {@link widget-components.FixedList|FixedList} component as tab bar.</caption>
 *    args => <FixedList {...args}>{args.children}</FixedList>
 *
 * @example <caption>Advanced usage of {@link widget-components.ScrolledList|ScrolledList} component as tab bar.</caption>
 *    args =>
 *       <ScrolledList
 *          {...args}
 *          alignItems={ScrolledList.ALIGN_ITEMS.LEFT}
 *          step={3}>
 *             {args.children}
 *       </ScrolledList>
 */

/**
 * @name TabPagination_RenderTabListArgs
 * @property {number} selected Initially selected list item (tab)
 * @property {function(number)} onItemClick Item clicked handler. Called with item's index argument.
 * @property {ReactElement[]} children Array of list items (tabs)
 */

/**
 * @property [children] {ReactElement[]} Tab content elements
 * @property [renderTab] {TabPagination_RenderTab} Function called in order to render single tab on tab bar. Renders tab index by default.
 * @property [renderTabList] {TabPagination_RenderTabList} Function called in order to render tab bar. Renders {@link widget-components.ScrolledList|ScrolledList} by default.
 * @property [selected=0] {number} Currently selected tab index
 */
TabPagination.propTypes = {
   children: PropTypes.node,
   renderTab: PropTypes.func,
   renderTabList: PropTypes.func,
   selected: PropTypes.number
};

TabPagination.defaultProps = {
   renderTab: idx => <div key={idx}><strong>{idx}</strong></div>,
   renderTabList: (args) => <ScrolledList {...args}>{args.children}</ScrolledList>,
   selected: 0
};

export default TabPagination;
