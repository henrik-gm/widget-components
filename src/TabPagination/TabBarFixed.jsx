import React, { Component, PropTypes } from 'react';
import styles from './TabBarFixed.scss';
import TabContainer from './TabContainer';

/**
 * Fixed width Tab Bar component
 */
class TabBarFixed extends Component {

   /**
    * Constructs.
    * @param {object} props Component properties
    */
   constructor(props) {
      super(props);
      this.state = {
         selected: props.selected
      };
   }

   /**
    * Switches to given tab.
    * @param {number} idx Tab index
    */
   onTabClick(idx) {
      if (this.props.onTabSwitch) {
         this.props.onTabSwitch(idx);
      }

      this.setState({ selected: idx });
   }

   /**
    * Renders component.
    * @returns {XML}
    */
   render() {
      return (
         <div className={styles.bar}>
            {this.props.children.map((child, i) => this.props.renderTabContainer({
               key: i,
               selected: this.state.selected == i,
               onClick: this.onTabClick.bind(this, i),
               children: child
            }))}
         </div>
      );
   }
}

TabBarFixed.propTypes = {
   /**
    * Tab element
    */
   children: PropTypes.arrayOf(PropTypes.element).isRequired,

   /**
    * Tab clicked handler
    */
   onTabSwitch: PropTypes.func,

   /**
    * Selected tab index
    */
   selected: PropTypes.number,

   /**
    * Function capable of rendering tab container
    */
   renderTabContainer: PropTypes.func
};

TabBarFixed.defaultProps = {
   selected: 0,
   renderTabContainer: (props) => <TabContainer {...props}>{props.children}</TabContainer>
};

export default TabBarFixed;
