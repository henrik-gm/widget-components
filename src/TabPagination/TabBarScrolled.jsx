import React, { Component, PropTypes } from 'react';
import styles from './TabBarScrolled.scss';
import ArrowButton from './ArrowButton';
import TabContainer from './TabContainer';

/**
 * Window resize handling reflex
 * @type {number}
 */
const RESIZE_SLUGGISHNESS = 200;

/**
 * Will turn off mobile mode if screen is wider than defined below
 * @type {number}
 */
const MOBILE_MAX_SCREEN_WIDTH = 768;

/**
 * Mobile browser check
 * @type {boolean}
 */
const isMobileBrowser = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);

/**
 * Scrolled Tab Bar component
 */
class TabBarScrolled extends Component {

   /**
    * Constructs.
    * @param {object} props Component properties
    */
   constructor(props) {
      super(props);

      this.state = {
         tab: props.selected
      };

      this.scrollLeft = 0;

      this.prevPage = this.prevPage.bind(this);
      this.nextPage = this.nextPage.bind(this);
      this.onResize = this.onResize.bind(this);
   }

   /**
    * Called after component rendering to DOM.
    */
   componentDidMount() {
      window.addEventListener('resize', this.onResize);
      this.scrollToTab(this.state.tab);
   }

   /**
    * Called before removing component.
    */
   componentWillUnmount() {
      window.removeEventListener('resize', this.onResize);
   }

   /**
    * Handles window resize
    */
   onResize() {
      if (this.resizeTimeout) {
         clearTimeout(this.resizeTimeout);
      }

      this.resizeTimeout = setTimeout(() => {
         this.scrollToTab(this.state.tab);
      }, RESIZE_SLUGGISHNESS);
   }

   /**
    * Selects tab with given index.
    * @param {number} idx Tab index
    */
   onTabClick(idx) {
      if (this.props.onTabSwitch) {
         this.props.onTabSwitch(idx);
      }

      this.scrollToTab(idx);

      this.setState({ tab: idx });
   }

   /**
    * Determines if component is running on mobile device.
    * @returns {boolean}
    */
   get mobileMode() {
      const hasTouchStart = 'ontouchstart' in window;

      if (!this.containerWidth) {
         return hasTouchStart && isMobileBrowser;
      }

      return this.containerWidth <= MOBILE_MAX_SCREEN_WIDTH && hasTouchStart && isMobileBrowser;
   }

   /**
    * Whole component's width.
    * @returns {number}
    */
   get containerWidth() {
      return this.container ? this.container.offsetWidth : null;
   }

   /**
    * Field of view width.
    * @returns {number|null}
    */
   get eyeshotWidth() {
      return this.eyeshot ? this.eyeshot.offsetWidth : null;
   }

   /**
    * Single tab width (in pixels)
    * @returns {number|null}
    */
   get tabWidth() {
      return this.firstTabWidth;
   }

   /**
    * Returns maximal scroll left offset.
    * @returns {number}
    */
   get maxScrollLeft() {
      if (!(this.tabWidth && this.eyeshotWidth)) {
         return null;
      }

      return this.props.children.length * this.tabWidth - this.eyeshotWidth;
   }

   /**
    * Scrolls bar to given offset.
    * @param {number} offset Scroll offset
    */
   scrollTo(offset) {
      if (!(this.bar && this.tabWidth && this.eyeshotWidth && this.eyeshot)) {
         return;
      }

      this.scrollLeft = offset > this.maxScrollLeft ? this.maxScrollLeft : offset;
      this.scrollLeft = this.scrollLeft < 0 ? 0 : this.scrollLeft;

      if (this.mobileMode) {
         this.eyeshot.scrollLeft = this.scrollLeft;
         this.translateX = 0;
      } else {
         this.translateX = -1 * this.scrollLeft;
         this.eyeshot.scrollLeft = 0;
      }

      this.forceUpdate();
   }

   /**
    * Translates bar element by given offset.
    * @param {number} offsetX X axis offset
    */
   set translateX(offsetX) {
      const transform = `translate3d(${offsetX}px, 0, 0)`;
      this.bar.style.transform = transform;
      this.bar.style.webkitTransform = transform;
      this.bar.style.mozTransform = transform;
   }

   /**
    * Scrolls bar to given tab.
    * @param {number} tab Tab index
    */
   scrollToTab(tab) {
      if (!(this.bar && this.tabWidth && this.eyeshotWidth)) {
         return;
      }

      this.scrollTo(tab * this.tabWidth - (this.eyeshotWidth - this.tabWidth) / 2);
   }

   /**
    * Should prev button be shown
    * @returns {boolean}
    */
   get showPrevButton() {
      return !this.mobileMode && this.scrollLeft > 0;
   }

   /**
    * Should next button be shown
    * @returns {boolean}
    */
   get showNextButton() {
      return !this.mobileMode && this.scrollLeft < this.maxScrollLeft;
   }

   /**
    * Scrolls tab bar to previous page.
    */
   prevPage() {
      this.scrollTo(this.scrollLeft - this.props.step * this.tabWidth);
   }

   /**
    * Scrolls tab bar to next page.
    */
   nextPage() {
      this.scrollTo(this.scrollLeft + this.props.step * this.tabWidth);
   }

   /**
    * Renders scrolled tab bar.
    * @returns {XML}
    */
   render() {
      return (
         <div className={styles.container} ref={el => (this.container = el)}>
            <div className={styles.eyeshot} ref={el => (this.eyeshot = el)}>
               <div className={styles.bar} ref={el => (this.bar = el)}>
                  {this.props.children.map((child, i) => this.props.renderTabContainer({
                     key: i,
                     selected: this.state.tab == i,
                     onClick: this.onTabClick.bind(this, i),
                     onWidth: i == 0 ? width => (this.firstTabWidth = width) : undefined,
                     children: child
                  }))}
               </div>
            </div>
            {this.props.renderPrevButton({
               onClick: this.prevPage,
               disabled: !this.showPrevButton
            })}
            {this.props.renderNextButton({
               onClick: this.nextPage,
               disabled: !this.showNextButton
            })}
         </div>
      );
   }
}

TabBarScrolled.propTypes = {
   /**
    * Tab element
    */
   children: PropTypes.arrayOf(PropTypes.element).isRequired,

   /**
    * Tab clicked handler
    */
   onTabSwitch: PropTypes.func,

   /**
    * Selected tab index
    */
   selected: PropTypes.number,

   /**
    * Scroll step (items count)
    */
   step: PropTypes.number,

   /**
    * Function capable of rendering button responsible for scrolling left
    */
   renderPrevButton: PropTypes.func,

   /**
    * Function capable of rendering button responsible for scrolling right
    */
   renderNextButton: PropTypes.func,

   /**
    * Function capable of rendering tab container
    */
   renderTabContainer: PropTypes.func
};

TabBarScrolled.defaultProps = {
   selected: 0,
   step: 2,
   renderPrevButton: (props) => <ArrowButton type='left' {...props} />,
   renderNextButton: (props) => <ArrowButton type='right' {...props} />,
   renderTabContainer: (props) => <TabContainer {...props}>{props.children}</TabContainer>
};

export default TabBarScrolled;
