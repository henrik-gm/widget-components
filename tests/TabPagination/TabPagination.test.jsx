import React, { Children } from 'react';
import TabPagination from '../../src/TabPagination/TabPagination';
import ReactTestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('TabPagination DOM rendering', () => {

   it('renders correctly with default props', () => {
      const tree = ReactTestRenderer.create(
         <TabPagination />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with one child item', () => {
      const tree = ReactTestRenderer.create(
         <TabPagination>
            <div />
         </TabPagination>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with many child items', () => {
      const tree = ReactTestRenderer.create(
         <TabPagination>
            <div>1</div>
            <div>2</div>
            {[<div key="3">3</div>, <div key="4">4</div>]}
         </TabPagination>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with selected item given arbitrary', () => {
      const tree = ReactTestRenderer.create(
         <TabPagination selected={1}>
            <div>1</div>
            <div>2</div>
         </TabPagination>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with custom renderTab func', () => {
      const renderTabMock = (i) => <div key={i}>Tab #{i}</div>;

      const tree = ReactTestRenderer.create(
         <TabPagination renderTab={renderTabMock}>
            <div>1</div>
            <div>2</div>
         </TabPagination>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with custom renderTabList func', () => {
      const renderTabListMock = ({selected, children}) =>
         <div>
            {Children.map(children, (child, i) =>
               <div className={selected == i ? 'sel' : 'not_sel'}>
                  {child}
               </div>)}
         </div>;

      const tree = ReactTestRenderer.create(
         <TabPagination renderTabList={renderTabListMock}>
            <div>1</div>
            <div>2</div>
         </TabPagination>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});

describe('TabPagination behaviour', () => {

   it('switches tab correctly with custom renderTab func', () => {
      const renderTabMock = jest.fn(i => <div key={i} className={`tabMock_${i}`}>{i}</div>);

      const wrapper = mount(
         <TabPagination
            renderTab={renderTabMock}
         >
            <div>1</div>
            <div>2</div>
         </TabPagination>
      );

      wrapper.find('.tabMock_1').simulate('click');
   });

});
