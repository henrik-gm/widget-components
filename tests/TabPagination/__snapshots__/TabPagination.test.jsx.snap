exports[`TabPagination DOM rendering renders correctly with custom renderTab func 1`] = `
<div>
  <div
    className="container ">
    <div
      className="eyeshot">
      <div
        className="bar">
        <div
          className="item selected"
          onClick={[Function]}>
          <div>
            Tab #
            0
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
        <div
          className="item "
          onClick={[Function]}>
          <div>
            Tab #
            1
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
      </div>
    </div>
    <button
      className="left"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
    <button
      className="right"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
  </div>
  <div>
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`TabPagination DOM rendering renders correctly with custom renderTabList func 1`] = `
<div>
  <div>
    <div
      className="sel">
      <div
        style={
          Object {
            "padding": 16,
          }
        }>
        <strong>
          0
        </strong>
      </div>
    </div>
    <div
      className="not_sel">
      <div
        style={
          Object {
            "padding": 16,
          }
        }>
        <strong>
          1
        </strong>
      </div>
    </div>
  </div>
  <div>
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`TabPagination DOM rendering renders correctly with default props 1`] = `
<div>
  <div
    className="container ">
    <div
      className="eyeshot">
      <div
        className="bar" />
    </div>
    <button
      className="left"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
    <button
      className="right"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
  </div>
  <div />
</div>
`;

exports[`TabPagination DOM rendering renders correctly with many child items 1`] = `
<div>
  <div
    className="container ">
    <div
      className="eyeshot">
      <div
        className="bar">
        <div
          className="item selected"
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              0
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
        <div
          className="item "
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              1
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
        <div
          className="item "
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              2
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
        <div
          className="item "
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              3
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
      </div>
    </div>
    <button
      className="left"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
    <button
      className="right"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
  </div>
  <div>
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`TabPagination DOM rendering renders correctly with one child item 1`] = `
<div>
  <div
    className="container ">
    <div
      className="eyeshot">
      <div
        className="bar">
        <div
          className="item selected"
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              0
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
      </div>
    </div>
    <button
      className="left"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
    <button
      className="right"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`TabPagination DOM rendering renders correctly with selected item given arbitrary 1`] = `
<div>
  <div
    className="container ">
    <div
      className="eyeshot">
      <div
        className="bar">
        <div
          className="item "
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              0
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
        <div
          className="item selected"
          onClick={[Function]}>
          <div
            style={
              Object {
                "padding": 16,
              }
            }>
            <strong>
              1
            </strong>
          </div>
          <i
            className="KambiWidget-primary-background-color border" />
        </div>
      </div>
    </div>
    <button
      className="left"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
    <button
      className="right"
      disabled={true}
      onClick={[Function]}>
      <svg
        className="icon"
        height="17px"
        version="1.1"
        viewBox="0 860 420 500"
        width="10px"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
          id="arrowLeft"
          transform="scale(1, -1) translate(0, -1500)" />
      </svg>
    </button>
  </div>
  <div>
    <div>
      2
    </div>
  </div>
</div>
`;
